### ISSUE TITLE
defaultIgnoreParams and ignoreParams are not working properly

### ISSUE THREAD

**Issue by pasviber** [nextflow-io/nf-schema#77]
Hi,

Recently, I finished developing a pipeline using nextflow (v24.04.3), but I am having problems with nf-schema (>=2.0.0). 

I have 2 conflicting values in the pipeline: pfam_release (Default value: “37.0”) and rnacentral_release (Default value: “24.0”). Both parameters are strings and so I make it known in the nextflow_schema.json putting as type string. However, when I run the pipeline putting as parameter for example --pfam_release 38.0 or --pfam_release “38.0”, it returns an error because it considers it as an integer and in the nextflow_schema.json it is a string.

```
ERROR ~ Validation of pipeline parameters failed!

 -- Check '.nextflow.log' file for details
The following invalid input values have been detected:

* --pfam_release (38.0): Value is [integer] but should be [string]
```

Therefore, I decided to use both the defaultIgnoreParams and ignoreParams validation parameters, as shown in this code extracted from my nextflow.config:

```
validation {
  defaultIgnoreParams = ["pfam_release", "rnacentral_release"]
  help {
    enabled = true
    command = "nextflow run main.nf -profile <docker/singularity/.../institute> --input samplesheet.csv --outdir <OUTDIR>"
  }
}
```

However, the error remains the same. So, **are defaultIgnoreParams and ignoreParams working properly?** @nvnieuwk 

lenientMode = true is not a solution for me because I don't want to be lax with all parameters. I just want those two not to be evaluated. A temporary solution is to remove from the nextflow_schema.json the line where the type of these two parameters is specified, but if I want to edit it again using nf-core pipelines schema build I get an error precisely because the type lines in those parameters have disappeared.

Thanks in advance

Pascual

**Comment by nvnieuwk** [comment]
Hi @pasviber 

Thank you for reporting this. Can you provide me some more information like what version of nf-schema and Nextflow you are using exactly?

**Comment by pasviber** [comment]
The nextflow version is v24.04.3, but in the case of nf-schema I have checked several, for example nf-schema@2.0.0, nf-schema@2.1.2 and nf-schema@2.2.0. In all versions I get the same problem. The problem does not seem to be linked to any particular version of nf-schema.

**Comment by pasviber** [comment]
To reproduce the error, using the command line, try entering the value 37.0 in a parameter (defined as a string in the nextflow_schema.json) of one of the pipelines you use. This will obviously return an error during parameter validation because it will find an integer where there should be a string. Then, try putting this parameter in the defaultIgnoreParams list and re-run the pipeline with the same command. If the error persists, something is wrong.

Thanks @nvnieuwk :)

**Comment by nvnieuwk** [comment]
Hi I've found the issue and have posted a fix in PR #78 :) I'll let you know once this is in a released version of the plugin
